src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1321,6 +1321,11 @@ private BadHttpRequestException GetInvalidRequestTargetException(ReadOnlySpan<by
                 ? target.GetAsciiStringEscaped(Constants.MaxExceptionDetailSize)
                 : string.Empty);
 
+    public void SetKeepAliveToClose()
```

Comment:
Nit:

```suggestion
    public void DisableHttp1KeepAlive()
```

It would be nice if we only called this for HTTP/1.x requests, but that would complicate matters, and this should basically be a no-op for `Http2Stream` and `Http3Stream` since both reset this value to true. @JamesNK

Reply:
I think is best to change to this:

```cs
    if (_httpVersion == Http.HttpVersion.Http10 || _httpVersion == Http.HttpVersion.Http11)
    {
        _keepAlive = false;
    }
```
or this:

```cs
    // prevent proxies of forwarding the connection header in case of http 1.0
    if (_httpVersion == Http.HttpVersion.Http11)
    {
        _keepAlive = false;
    }
```

see [rfc2068 19.7.1](https://datatracker.ietf.org/doc/html/rfc2068#section-19.7.1) and [rfc2068 14.10](https://datatracker.ietf.org/doc/html/rfc2616/#section-14.10)